kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-common/src/main/java/org/kie/workbench/common/dmn/webapp/kogito/common/client/converters/DMNMarshallerImportsHelperKogitoImpl.java

```diff
@@ -16,29 +16,256 @@
+        this.contentService = contentService;
+        this.diagramService = diagramService;
+        this.promises = promises;
+        this.diagramUtils = diagramUtils;
+        this.includedModelFactory = includedModelFactory;
+    }
+
+    @Override
+    public Promise<Map<JSITImport, JSITDefinitions>> getImportDefinitionsAsync(final Metadata metadata,
+                                                                               final List<JSITImport> imports) {
+        if (!imports.isEmpty()) {
+            return loadDMNDefinitions().then(otherDefinitions -> {
+                final Map<JSITImport, JSITDefinitions> importDefinitions = new HashMap<>();
+                for (final Map.Entry<String, JSITDefinitions> entry : otherDefinitions.entrySet()) {
+                    final JSITDefinitions def = Js.uncheckedCast(entry.getValue());
+                    findImportByDefinitions(def, imports).ifPresent(anImport -> {
+                        final JSITImport foundImported = Js.uncheckedCast(anImport);
+                        importDefinitions.put(foundImported, def);
+                    });
+                }
+                return promises.resolve(importDefinitions);
+            });
+        }
+        return promises.resolve(Collections.emptyMap());
+    }
+
+    Promise<Map<String, JSITDefinitions>> loadDMNDefinitions() {
+        return contentService.getFilteredItems(DMN_FILES_PATTERN, ResourceListOptions.assetFolder())
+                .then(list -> {
+                    if (list.length == 0) {
+                        return promises.resolve(Collections.emptyMap());
+                    } else {
+                        final Map<String, JSITDefinitions> otherDefinitions = new ConcurrentHashMap<>();
+                        return promises.all(Arrays.asList(list),
+                                            (String file) -> loadDefinitionFromFile(file, otherDefinitions))
+                                .then(v -> promises.resolve(otherDefinitions));
+                    }
+                });
+    }
+
+    @Override
+    public void loadNodesFromModels(final List<DMNIncludedModel> includedModels,
+                                    final ServiceCallback<List<DMNIncludedNode>> callback) {
+        final List<DMNIncludedNode> result = new Vector<>();
+        if (includedModels.isEmpty()) {
+            callback.onSuccess(result);
+        } else {
+            loadDMNDefinitions()
+                    .then(existingDefinitions -> promises.all(includedModels, model -> loadNodes(existingDefinitions, model, result))
+                            .then(p ->
+                                  {
+                                      callback.onSuccess(result);
+                                      return promises.resolve();
+                                  }));
+        }
+    }
+
+    private Promise loadNodes(final Map<String, JSITDefinitions> existingDefinitions,
```

Comment:
IntelliJ is warning about this method, I think we can solve it by using:
```suggestion
    private Promise<List<DMNIncludedNode>> loadNodes(final Map<String, JSITDefinitions> existingDefinitions,
```